DSharpPlus.Analyzers/Core/DSP0001/OptionalAddJsonIgnoreCodeFixProvider.cs

```diff
@@ -0,0 +1,115 @@
+            PropertyDeclarationSyntax declaration = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<PropertyDeclarationSyntax>().First();
+            context.RegisterCodeFix(CodeAction.Create(
+                title: CodeFixTitle,
+                createChangedDocument: c => AddJsonIgnoreAttributeAsync(context.Document, declaration, c),
+                equivalenceKey: CodeFixTitle), diagnostic);
+        }
+
+        private static async Task<Document> AddJsonIgnoreAttributeAsync(Document document, PropertyDeclarationSyntax propertyDeclarationSyntax, CancellationToken cancellationToken)
+        {
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken);
+            Document newDocument = AddAttributes(document, root, propertyDeclarationSyntax);
+            return newDocument.WithSyntaxRoot(await AddUsings((await newDocument.GetSyntaxRootAsync(cancellationToken)).SyntaxTree.GetCompilationUnitRoot(cancellationToken)).GetRootAsync(cancellationToken));
+        }
+
+        private static Document AddAttributes(Document document, SyntaxNode root, PropertyDeclarationSyntax propertyDeclarationSyntax)
+        {
+            AttributeSyntax jsonIgnoreAttribute = SyntaxFactory
+                .Attribute(SyntaxFactory.IdentifierName("JsonIgnore"))
+                .WithArgumentList(SyntaxFactory.AttributeArgumentList(SyntaxFactory.SingletonSeparatedList(SyntaxFactory
+                    .AttributeArgument(SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, SyntaxFactory.IdentifierName("JsonIgnoreCondition"), SyntaxFactory.IdentifierName("WhenWritingDefault")))
+                    .WithNameEquals(SyntaxFactory.NameEquals(SyntaxFactory.IdentifierName("Condition"))))));
+
+            // We are moving trivia
+            //AttributeListSyntax firstAttribute = propertyDeclarationSyntax.AttributeLists.FirstOrDefault();
+            //if (firstAttribute != null)
+            //{
+            //    // All this just to preserve trivia (XML docs)
+            //    jsonIgnoreAttribute = jsonIgnoreAttribute.WithTriviaFrom(firstAttribute);
+            //    return document.WithSyntaxRoot(root.ReplaceNode(propertyDeclarationSyntax, propertyDeclarationSyntax.WithAttributeLists(propertyDeclarationSyntax.AttributeLists.Replace(firstAttribute, firstAttribute.WithoutTrivia()).Insert(0, SyntaxFactory.AttributeList(SyntaxFactory.SingletonSeparatedList(jsonIgnoreAttribute))))));
+            //}
+            //else
+            //{
+            //}
+#pragma warning disable IDE0046 // Convert to conditional expression
+            if (propertyDeclarationSyntax.AttributeLists.Any())
+            {
+                return document.WithSyntaxRoot(root.ReplaceNode(propertyDeclarationSyntax, propertyDeclarationSyntax.WithAttributeLists(propertyDeclarationSyntax.AttributeLists.Add(SyntaxFactory.AttributeList(SyntaxFactory.SingletonSeparatedList(jsonIgnoreAttribute))))));
```

Comment:
The expression `SyntaxFactory.AttributeList(SyntaxFactory.SingletonSeparatedList(jsonIgnoreAttribute))` seems to be present in each of these branches. That should probably be factored out. Adding some appropriate line breaks would also help with readability.

Reply:
Looking at the current commit, how would you think I should go about this?

Reply:
Will do! Thanks for your input